### PR TITLE
avm2: Small XML fixes

### PR DESCRIPTION
**Pull request by sleepycatcoding**

### CODE REVIEW

File: core/src/avm2/object/xml_list_object.rs [hunk 1]
```diff
@@ -135,6 +137,18 @@ impl<'gc> XmlListObject<'gc> {
         )
     }
 
+    pub fn to_xml_string(&self, activation: &mut Activation<'_, 'gc>) -> AvmString<'gc> {
```
Comment (torokati44) [inline core/src/avm2/object/xml_list_object.rs:140]: I guess clippy would like this better as `as_xml_string`...

Comment (sleepycatcoding) [reply core/src/avm2/object/xml_list_object.rs:140]: Should I rename this or allow the lint?

Comment (torokati44) [reply core/src/avm2/object/xml_list_object.rs:140]: I think renaming it is fine.

File: core/src/avm2/globals/xml.rs [hunk 2]
```diff
@@ -600,14 +599,22 @@ pub fn insert_child_after<'gc>(
     }
 
     // 3. Else if Type(child1) is XML
-    if let Some(child1) = child1.and_then(|x| x.as_xml_object()) {
+    if let Some(child1) = child1.as_object().and_then(|x| {
+        if let Some(xml) = x.as_xml_object() {
+            return Some(*xml.node());
+        } else if let Some(list) = x.as_xml_list_object() {
```
Comment (evilpie) [inline core/src/avm2/globals/xml.rs:605]: Please add a note that this is non-standard avmplus behavior.

Comment (sleepycatcoding) [reply core/src/avm2/globals/xml.rs:606]: Added.

File: core/src/avm2/object/xml_list_object.rs (hunk 1 above)
Comment (sleepycatcoding) [reply core/src/avm2/object/xml_list_object.rs]: Renamed.

**Review by Dinnerbone — APPROVED** [review]

**Pull request merged**